drivers/video/vnc/vnc_fbdev.c

```diff
@@ -470,6 +472,42 @@ static int up_waitforsync(FAR struct fb_vtable_s *vtable)
 }
 #endif
 
+#ifdef CONFIG_VNCSERVER_MOUSE
+
+/****************************************************************************
+ * Name: vnc_uinput_mousein
+ ****************************************************************************/
+
+static int vnc_uinput_mouseout(FAR void *arg, int16_t x,
+                               int16_t y, uint8_t buttons)
+{
+  FAR struct vnc_session_s *session = arg;
+  struct touch_sample_s sample;
+
+  DEBUGASSERT(session);
+
+  sample.npoints = 1;
+  sample.point[0].x = x;
+  sample.point[0].y = y;
+
+  if ((buttons & MOUSE_BUTTON_1) != 0)
+    {
+      sample.point[0].flags = TOUCH_DOWN |
+                              TOUCH_ID_VALID |
+                              TOUCH_POS_VALID;
+    }
+  else
+    {
+      sample.point[0].flags = TOUCH_UP |
+                              TOUCH_ID_VALID |
+                              TOUCH_POS_VALID;
+    }
+
+  return nx_write(session->fd_mouse, (void *)&sample, sizeof(sample));
```

Comment:
why not implement and register a mouse driver directly?

Reply:
This causes VNC mouse driver's private data and VNC session to be coupled together.
And write to uinput is more simple (it's logic is very similar to uinput).

Reply:
But touch screen driver support the upper/lower half in the recent update, which mean writing a touch screen driver isn't complex than send the event through uinput.

Reply:
OK, a touchscreen based driver implemented.